docs/adrs/0000-composite-actions.md

```diff
@@ -0,0 +1,97 @@
+**Date**: 2021-06-10
+
+**Status**: Accepted
+
+## Context
+
+We released [composite run steps](https://github.com/actions/runner/pull/554) last year which started our journey of reusing steps across different workflow files. To continue that journey, we want to expand composite run steps into composite actions.
+
+We want to support the `uses` steps from workflows in composite actions, including:
+  - Container actions
+  - javascript actions
+  - and even other Composite actions (up to a limit of course!)
+  - The pre and post steps these actions can generate
+
+## Guiding Principles
+
+- Composite Actions should function as a single step or action, no matter how many steps it is composed of or how many levels of recursion it has
```

Comment:
If we are pushing composite actions as a general step re-use mechanism, it seems like users will want an option to control whether everything is collapsed into a single step log. For now of course, let's just continue down the path of collapsing into a single step log. Simply stating that I anticipate making composite actions more useful (support all step types) will to lead to follow-up work to control the collapse behavior.